keras_nlp/layers/sine_positional_embedding.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sinusoidal position embedding layer."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class SinePositionEncoding(keras.layers.Layer):
+    """Sinusoidal positional encoding layer.
+
+    This layer calculates the position encoding as a mix of sine and cosine
+    functions with geometrically increasing wavelengths. Defined and formulized
+    in [Attention is All You Need](https://arxiv.org/abs/1706.03762).
+
+    Usage:
+        Takes as input an embedded token tensor, where the last dimension is
+        assumed to be the feature dim, and the second to last dimension is
+        assumed to be the sequence dimension. This layer will return a
+        positional encoding the same size as the embedded token tensor, which
+        can be added directly to the embedded token tensor.
+
+    Args:
+        min_frequency: The minimum frequency for each position embedding,
+                       initalized to 1.0e-4.
+
+    Example:
+    ```python
+    # create a simple embedding layer with sinusoidal positional encoding
+    inputs = keras.Input((100, ), dtype=tf.float32)
+    embedding = keras.layers.Embedding(input_dim=1000, output_dim=32)(inputs)
+    positional_encoding = keras_nlp.layers.SinePositionEncoding()(embedding)
+    outputs = embedding + positional_encoding
+    ```
+
+    References:
+      [Attention is All You Need](https://arxiv.org/abs/1706.03762)
+    """
+
+    def __init__(
+        self,
+        min_frequency: float = 1.0e-4,
```

Comment:
Somehow I cannot reply to the previous discussion thread. 

I mean when your `min_frequency` arg is <1, isn't it the max_frequency? the frequency formula is (0.0001) ** (2i / d_model). Also this min/max is very unstable, because it is controlled by the actual value. so I would prefer renaming it to something like `base_frequency`.

Reply:
2i/d_model is less than 1, so the value will increase. But still, base_frequency is much better.

Reply:
Yea, my mistake, I did not notice the < 1 constraint.